mlibrary/search
# [LIBSEARCH-878] Fix accessibility errors and warnings (Part 2)

# Overview
An [accessibility evaluation](https://docs.google.com/document/d/1LiRYT1Q_eaN4mJuOb70QIVHKemx4khs7cV19JCHynBY/edit?usp=sharing) was done on Search. This pull request resolves the violations and warnings that were found to the original `Modal` component. The `Dialog` component has been created to replace it, using the native HTML `dialog` element. All components except for `ChooseAffiliation` have been moved outside of the `Dialog` component.

This pull request relates to [LIBSEARCH-878](https://mlit.atlassian.net/browse/LIBSEARCH-878).

## Anything else?
The issues relating to the accordions are not included in this, as they require building a new component.

## Testing
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- Go through the site to see if anything is broken.
- Choose affiliation:
  - Can you select a new one?
  - Will the selection appear active?
- Actions list:
  - Can you copy the link?
  - Can you copy citations?
